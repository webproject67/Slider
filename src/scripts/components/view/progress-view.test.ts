import ProgressView from './progress-view';
import Model from '../model/model';

const state = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 0,
  progress: true,
  range: 'one',
  draft: 0,
  start: 1,
  configuring: 0,
  scale: true,
  step: 1,
  to: -10000,
  toPercent: 100,
  view: 'horizontal',
};

const model = new Model(state);
const progressView = new ProgressView();

describe('snapshot', () => {
  test('renders correctly element1', () => {
    expect(new ProgressView().getElement(state)).toMatchSnapshot();
  });

  test('renders correctly element2', () => {
    model.setValue(['view'], ['vertical']);
    expect(new ProgressView().getElement(state)).toMatchSnapshot();
  });
});

describe('handleBarClick', () => {
  test('spyOn bar click', () => {
    const somethingSpy = jest.spyOn(progressView, 'handleBarClick');
    let evt: any;
    progressView.handleBarClick(state, evt);
    expect(somethingSpy).toHaveBeenCalledTimes(1);
  });
});
