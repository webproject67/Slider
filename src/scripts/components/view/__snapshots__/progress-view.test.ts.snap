// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly element1 1`] = `
<div
  class="slider__bars"
>
  <div
    class="slider__bar"
    style="margin-left:0%;margin-right:0%"
  />
</div>
`;

exports[`snapshot renders correctly element2 1`] = `
<div
  class="slider__bars"
>
  <div
    class="slider__bar slider__bar_with-width"
    style="top:0%;height:100%"
  />
</div>
`;
